### PR TITLE
process: api to set file descriptor soft limit

### LINKED ISSUE #2584: OS X: Max Files Limit
I updated my system-wide max files limit on OS X via a plist in Library/LaunchDaemons to be 10240, and have confirmed this using `launchctl limit`.

But I found Electron is still getting EMFILE at around 1024 descriptors.

I think this behavior is inherited from Chrome, which is blindly setting its own limit, probably with the intention of increasing the very low usual 256 limit on Mac OS X, but now actually getting something less than the system limit.

Would it be possible to increase it to 10240 or provide a function to change the ulimit on OS X or Linux? I would not like to have to bundle a native add-on with Electron to do this, and Node.js does not yet provide access to the posix methods for changing ulimit.

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #2584

### CODE REVIEW

**Comment by hokein** [comment]
Does it only make effect on `main process`?

**Comment by deepak1556** [comment]
> Does it only make effect on main process?

was under the impression the render process would inherit main process `ulimit`, was wrong in the end. Have fixed it, thanks @hokein !

**Comment by zcbenz** [comment]
I think we can just make it a method of `process` object, so users can just call it as they need. Under most cases users only need to set fd limit for workers or only renderer process.

**Comment by deepak1556** [comment]
@zcbenz Fixed, thanks!

File: docs/api/process.md [hunk 1]
```diff
@@ -11,3 +11,10 @@ upstream node:
 ## process.hang
 
 Causes the main thread of the current process hang.
+
+## process.setFdLimit(maxDescriptors) _OS X_ _Linux_
+
+* `maxDescriptors` Number
```
Comment (zcbenz) [inline docs/api/process.md]: Should be `Integer` here.

Comment (deepak1556) [inline docs/api/process.md]: Fixed.

**Comment by zcbenz** [comment]
:+1:

**Comment by jorangreef** [comment]
Thanks @deepak1556!

**Pull request merged**